src/Containers/DenseAxisArray.jl

```diff
@@ -559,3 +574,70 @@ end
 # but some users may depend on it's functionality so we have a work-around
 # instead of just breaking code.
 Base.repeat(x::DenseAxisArray; kwargs...) = repeat(x.data; kwargs...)
+
+###
+### view
+###
+
+_get_subaxis(::Colon, b) = b
+
+function _get_subaxis(a::AbstractVector, b)
+    for ai in a
+        if !(ai in b)
+            throw(KeyError(ai))
+        end
+    end
+    return a
+end
+
+function _get_subaxis(a::T, b::AbstractVector{T}) where {T}
+    if !(a in b)
+        throw(KeyError(a))
+    end
+    return a
+end
+
+struct DenseAxisArrayView{T,N,D,A} <: AbstractArray{T,N}
+    data::D
+    axes::A
+    function DenseAxisArrayView(
+        x::Containers.DenseAxisArray{T,N},
+        args...,
+    ) where {T,N}
+        axis = _get_subaxis.(args, axes(x))
+        return new{T,N,typeof(x),typeof(axis)}(x, axis)
+    end
+end
+
+function Base.view(A::Containers.DenseAxisArray, args...)
+    return DenseAxisArrayView(A, args...)
+end
+
+Base.size(x::DenseAxisArrayView) = length.(x.axes)
+
+Base.axes(x::DenseAxisArrayView) = x.axes
+
+function Base.getindex(x::DenseAxisArrayView, args...)
+    y = _get_subaxis.(args, x.axes)
+    return getindex(x.data, y...)
+end
+
+Base.getindex(a::DenseAxisArrayView, k::DenseAxisArrayKey) = a[k.I...]
+
+function Base.setindex!(x::DenseAxisArrayView, args...)
+    return setindex!(x.data, args...)
+end
+
+function Base.eachindex(A::DenseAxisArrayView)
+    # Return a generator so that we lazily evaluate the product instead of
+    # collecting into a vector.
+    return (DenseAxisArrayKey(k) for k in Base.product(A.axes...))
```

Comment:
The access provided is still slower since we need to do the dictionary `getindex` instead of getting directly the elements of the underlying array.
It would be more efficient to return an iterator over the `CartesianIndex` directly by looking at the subset of each axes of `x.data` that we have.

Reply:
> It would be more efficient to return an iterator over the CartesianIndex directly by looking at the subset of each axes of x.data that we have.

But that would require similar lookups, just in `eachindex` instead of `getindex`? The view does not need to be contiguous. Although I guess it'd only need `sum(length.(A.axes))` instead of `length(A)` lookups.

I think this is a premature optimization. Let's see if anyone complains first.

Reply:
> Although I guess it'd only need `sum(length.(A.axes))` instead of `length(A)` lookups.

Yes exactly

> I think this is a premature optimization. Let's see if anyone complains first.

I agree but I would leave a comment in the code